### DISCUSSION TITLE
How to hide `Schemas` section in `/docs`?

### DISCUSSION THREAD

**Discussion by MarioIshac** [fastapi/fastapi#8993] [Questions] [answered]
```
from fastapi import FastAPI
from pydantic.main import BaseModel

app = FastAPI()

class Model(BaseModel):
    a: str

@app.get("/", response_model=BaseModel)
def get():
    return {"a": "b"}
```

After `uvicorn app:app`, `/docs` shows [this](https://i.imgur.com/KBcYdOF.png):

Is there any way to hide `Schemas` section with the `BaseModel`? I don't want to remove the schema definitions from the API spec. Instead, I just want to remove them from the render. I've checked out [Extending OpenAPI](https://fastapi.tiangolo.com/advanced/extending-openapi/), but didn't find anything related.

I found [How to hide the Models section in Swagger UI?](https://stackoverflow.com/questions/57138564/how-to-hide-the-models-section-in-swagger-ui) which mentions a field called `defaultModelsExpandDepth`, but not sure how I could configure such a thing through FastAPI.

**Comment by Mause** [comment]
You're using response_model wrong, it should be

```python

@app.get("/", response_model=Model)
def get():
    return {"a": "b"}
```

**Comment by MarioIshac** [comment]
> You're using response_model wrong, it should be
> 
> ```python
> @app.get("/", response_model=Model)
> def get():
>     return {"a": "b"}
> ```

Ah yeah my bad, meant to put the subclass. So question becomes, how would I hide the `Schemas` section holding `Model` instead of `BaseModel` that is shown in the image.

**Comment by MarioIshac** [comment]
I found out how to do this, I had to add `defaultModelsExpandDepth: -1,` (like the SO answers say) to https://github.com/tiangolo/fastapi/blob/master/fastapi/openapi/docs.py#L40, and then serve the custom generated HTML myself through https://fastapi.tiangolo.com/advanced/extending-openapi/#serve-the-static-files. Will open formal merge request for this soon, and close issue around then.

**Comment by bfontaine** [comment]
@MarioIshac did you end up working on it?

**Comment by jacobhjkim** [comment]
https://github.com/tiangolo/fastapi/pull/3168
This PR will take care of this issue.

**Comment by KYJiya** [comment]
> I found out how to do this, I had to add `defaultModelsExpandDepth: -1,` (like the SO answers say) to https://github.com/tiangolo/fastapi/blob/master/fastapi/openapi/docs.py#L40, and then serve the custom generated HTML myself through https://fastapi.tiangolo.com/advanced/extending-openapi/#serve-the-static-files. Will open formal merge request for this soon, and close issue around then.

it works. thank's.

**Comment by fdegier** [comment, marked as answer]
FastAPI now supports passing swagger parameters, so to hide the schemas you can do it as below:

```
app = FastAPI(
    title="API",
    swagger_ui_parameters={"defaultModelsExpandDepth": -1}
)
```

**Reply by qqlww1987** [reply]
Setting does not take effect

**Reply by ewertones** [reply]
@qqlww1987 I had the same problem.

In my case, I had  `docs_url=None` when initializing FastAPI, so these swagger parameters weren't being sent to `get_swagger_ui_html` due to this `if` condition:

https://github.com/tiangolo/fastapi/blob/3425c834cc113060d00b8a295d3456eb306a109d/fastapi/applications.py#L1012-L1026

I did that to remove `- Swagger UI` from the page title.

To solve my problem, I changed the `/docs` route to `get_swagger_ui_html` passing that config.

```python
@router.get("/docs", include_in_schema=False)
async def overriden_swagger():
    return get_swagger_ui_html(
        openapi_url="/openapi.json",
        title="Title",
        swagger_favicon_url="favicon.ico",
        swagger_ui_parameters={"defaultModelsExpandDepth": -1},
    )
```
